src/Bicep.LangServer.IntegrationTests/HoverTests.cs

```diff
@@ -89,11 +89,24 @@ public async Task HoveringOverSymbolReferencesAndDeclarationsShouldProduceHovers
 
             foreach (var symbolReference in symbolReferences)
             {
+                // Verify hover information is not available on keywords
+                if (symbolReference is ITopLevelDeclarationSyntax topLevelDeclarationSyntax)
+                {
+                    var hoverOnKeyword = await helper.Client.RequestHover(new HoverParams
+                    {
+                        TextDocument = new TextDocumentIdentifier(uri),
+                        Position = TextCoordinateConverter.GetPosition(lineStarts, topLevelDeclarationSyntax.Keyword.Span.Position)
+                    });
+
+                    hoverOnKeyword.Should().BeNull();
+
+                    continue;
+                }
+
                 // by default, request a hover on the first character of the syntax, but for certain syntaxes, this doesn't make sense.
                 // for example on an instance function call 'az.resourceGroup()', it only makes sense to request a hover on the 3rd character.
                 var nodeForHover = symbolReference switch
                 {
-                    ITopLevelDeclarationSyntax d => d.Keyword,
```

Comment:
Rather than removing this check, let's update it so that we continue to verify hovers are working properly - to match the current expected behavior, it would make sense to instead do 
```
ITopLevelDeclarationSyntax d => d.Name,
```